src/EditorFeatures/Core/Extensibility/NavigationBar/AbstractEditorNavigationBarItemService.cs

```diff
@@ -28,16 +35,17 @@ public async Task<IList<NavigationBarItem>> GetItemsAsync(Document document, Can
             return items.SelectAsArray(v => (NavigationBarItem)new WrappedNavigationBarItem(v));
         }
 
-        public void NavigateToItem(Document document, NavigationBarItem item, ITextView textView, CancellationToken cancellationToken)
-            => NavigateToItem(document, (WrappedNavigationBarItem)item, textView, cancellationToken);
+        public Task NavigateToItemAsync(Document document, NavigationBarItem item, ITextView textView, CancellationToken cancellationToken)
+            => NavigateToItemAsync(document, (WrappedNavigationBarItem)item, textView, cancellationToken);
 
-        protected void NavigateToSymbolItem(
+        protected async Task NavigateToSymbolItemAsync(
             Document document, RoslynNavigationBarItem.SymbolItem item, CancellationToken cancellationToken)
         {
             Contract.ThrowIfFalse(item.Kind == RoslynNavigationBarItemKind.Symbol);
             var symbolNavigationService = document.Project.Solution.Workspace.Services.GetRequiredService<ISymbolNavigationService>();
 
-            var symbolInfo = item.NavigationSymbolId.Resolve(document.Project.GetRequiredCompilationAsync(cancellationToken).WaitAndGetResult(cancellationToken), ignoreAssemblyKey: true, cancellationToken: cancellationToken);
```

Comment:
removal of WaitAndGetResults/WAit/GetXXXSynchronously entirely from navbar code (both controller and services).